### DISCUSSION TITLE
Trainer cannot find available GPUS

### DISCUSSION THREAD

**Discussion by omrishac** [Lightning-AI/pytorch-lightning#8705] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hey, 

I am trying to use Huyvnphan's implementation of CIFAR-10 classifier with Pytorch on Colab.  
In his implementation he uses pytorch-lightning and for some reason when I try to train the model PL raises an error: 
**"pytorch_lightning.utilities.exceptions.MisconfigurationException: GPUs requested but none are available."** 

note that when I run "torch.cuda.is_available()" I get True and "torch.cuda.device_count()" is 1.

The Trainer call looks like that: 

        trainer = Trainer(
            fast_dev_run=bool(args.dev),
            logger=logger if not bool(args.dev + args.test_phase) else None,
            gpus=-1,
            deterministic=True,
            weights_summary=None,
            log_every_n_steps=1,
            max_epochs=args.max_epochs,
            checkpoint_callback=checkpoint,
            precision=args.precision,
        )

please look at his implementation for more info: 
https://github.com/huyvnphan/PyTorch_CIFAR10

Thanks in advance,
OS

**Comment by lilylizhengzhou** [comment]
have exactly same issue when I try to run this repo

**Comment by akihironitta** [comment]
@lilylizhengzhou Lightning uses this simple condition to check the cuda device availability (in master):
https://github.com/Lightning-AI/lightning/blob/4508c7004c0f8075e556b6c6678c60b0e958719e/src/lightning_lite/accelerators/cuda.py#L100-L122

I'd recommend recreating your environment and installing PyTorch and Lightning again.
